### PR TITLE
Monad transformer instances for various type classes in `io-classes`

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

File: io-classes/src/Control/Monad/Class/MonadSTM.hs [hunk 1]
```diff
@@ -600,3 +609,66 @@ throwSTM = MonadThrow.throwIO
...
+    peekTBQueue    = ReaderTSTM  . peekTBQueue
+    tryPeekTBQueue = ReaderTSTM  . tryPeekTBQueue
+    flushTBQueue   = ReaderTSTM  . flushTBQueue
+    writeTBQueue   = ReaderTSTM |. writeTBQueue
+    lengthTBQueue  = ReaderTSTM  . lengthTBQueue
+    isEmptyTBQueue = ReaderTSTM  . isEmptyTBQueue
+    isFullTBQueue  = ReaderTSTM  . isFullTBQueue
+
+
+(|.) :: (a -> b) -> (c -> d -> a) -> (c -> d -> b)
```
Comment (rvl) [inline io-classes/src/Control/Monad/Class/MonadSTM.hs:672]: Cool

**Review by rvl — APPROVED** [review]
Looks pretty useful to me, thanks @coot.

**Review by bolt12 — APPROVED** [review]
Looks good to me :)

**Review by karknu — APPROVED** [review]
LGTM

**Comment by coot** [comment]
bors merge

**Pull request merged**